File: Insert Interval.js
var insert = function(intervals, newInterval) {
    let merged = [];

    for (let i=0; i<intervals.length; i++){
        if (newInterval[1]<intervals[i][0]){
            merged.push(newInterval);
            return merged.concat(intervals.slice(i));
        }
        else if (newInterval[0]>intervals[i][1]){
            merged.push(intervals[i]);
        }
        else {newInterval = [Math.min(intervals[i][0],newInterval[0]),Math.max(intervals[i][1],newInterval[1])]}
    }
    merged.push(newInterval) ; 
    return merged ; 
};

File: Insert Interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        merged = []
        
        for x in range(len(intervals)):
            if intervals[x][1]<newInterval[0]:
                merged.append(intervals[x])
            elif newInterval[1] < intervals[x][0]:
                
                merged.append(newInterval)
                return merged+intervals[x:]
            else:
                newInterval = [min(newInterval[0],intervals[x][0]),max(newInterval[1],intervals[x][1])]
        merged.append(newInterval) 
        return merged   